db/views/reporting_patient_states_per_month_v01.sql

```diff
@@ -0,0 +1,117 @@
+    assigned_facility.block_region_id as assigned_block_region_id,
+    assigned_facility.district_slug as assigned_district_slug,
+    assigned_facility.district_region_id as assigned_district_region_id,
+    assigned_facility.state_slug as assigned_state_slug,
+    assigned_facility.state_region_id as assigned_state_region_id,
+    assigned_facility.organization_slug as assigned_organization_slug,
+    assigned_facility.organization_region_id as assigned_organization_region_id,
+
+    ------------------------------------------------------------
+    -- information on registration facility and parent regions
+    p.registration_facility_id AS patient_registration_facility_id,
+    registration_facility.facility_slug as registration_facility_slug,
+    registration_facility.facility_region_id as registration_facility_region_id,
+    registration_facility.block_slug as registration_block_slug,
+    registration_facility.block_region_id as registration_block_region_id,
+    registration_facility.district_slug as registration_district_slug,
+    registration_facility.district_region_id as registration_district_region_id,
+    registration_facility.state_slug as registration_state_slug,
+    registration_facility.state_region_id as registration_state_region_id,
+    registration_facility.organization_slug as registration_organization_slug,
+    registration_facility.organization_region_id as registration_organization_region_id,
+
+    ------------------------------------------------------------
+    -- latest visit info for the month
+    bps.blood_pressure_recorded_at AS bp_recorded_at,
+    bps.systolic,
+    bps.diastolic,
+    visits.visited_at AS visited_at,
+    p.recorded_at AT TIME ZONE 'utc' AT TIME ZONE (SELECT current_setting('TIMEZONE')) as recorded_at,
+
+
+    (cal.year - DATE_PART('year', p.recorded_at AT TIME ZONE 'utc' AT TIME ZONE (SELECT current_setting('TIMEZONE')))) * 12 +
+    (cal.month - DATE_PART('month', p.recorded_at AT TIME ZONE 'utc' AT TIME ZONE (SELECT current_setting('TIMEZONE'))))
+    AS months_since_registration,
+
+    visits.months_since_visit AS months_since_visit,
+    bps.months_since_bp_observation AS months_since_bp,
+
+    ------------------------------------------------------------
+    -- categorization
+    CASE
+        WHEN (bps.systolic IS NULL OR bps.diastolic IS NULL) THEN 'unknown'
+        WHEN (bps.systolic < 140 AND bps.diastolic < 90) THEN 'controlled'
+        ELSE 'uncontrolled'
+        END
+        AS last_bp_state,
+
+    CASE
+        WHEN p.status = 'dead' THEN 'died'
+        WHEN (
+          -- months_since_registration
+          (cal.year - DATE_PART('year', p.recorded_at AT TIME ZONE 'utc' AT TIME ZONE (SELECT current_setting('TIMEZONE')))) * 12 +
+          (cal.month - DATE_PART('month', p.recorded_at AT TIME ZONE 'utc' AT TIME ZONE (SELECT current_setting('TIMEZONE')))) < 12
+
+          OR
+
+          bps.months_since_bp_observation < 12
+        ) THEN 'under_care'
```

Comment:
We considered doing this, then decided to leave it as is since it doesn't cause harm. And it might actually be a bit wonky to have nulls for patients that are not hypertensive.

Separately, we also considered reporting only on hypertensive patients since the ontology starts with them. But we might care about totals (including non hypertensive patients) in some places, and it doesn't cost us much to have it here, just an additional "hypertension = 'yes'" in most queries, which is probably fine.

Reply:
Can totally revisit this if we find that we're just repeating that clause everywhere.

Reply:
The prefix of `htn_` sounds good, I'll pick that up.

Reply:
Done in be7634a87fb8af6198219740ab73f7f042008e25.

Reply:
> Separately, we also considered reporting only on hypertensive patients since the ontology starts with them. But we might care about totals (including non hypertensive patients) in some places, and it doesn't cost us much to have it here, just an additional "hypertension = 'yes'" in most queries, which is probably fine.

Makes sense to me. We may need to rename some fields down the road to be more explicit about `htn` versus other diagnosis, but that isn't a big deal.